src/modalities/checkpointing/checkpoint_conversion.py

```diff
@@ -0,0 +1,70 @@
+import logging
+from pathlib import Path
+
+import torch
+from pydantic import BaseModel
+
+from modalities.checkpointing.checkpointing_execution import PytorchToDiscCheckpointing
+from modalities.config.component_factory import ComponentFactory
+from modalities.config.config import GPT2HuggingFaceAdapterConfig, PydanticModelIFType, load_app_config_dict
+from modalities.models.gpt2.gpt2_model import GPT2LLMConfig
+from modalities.models.gpt2.huggingface_model import HuggingFaceModel
+from modalities.registry.components import COMPONENTS
+from modalities.registry.registry import Registry
+
+
+class CheckpointConversion:
+    def __init__(self, checkpoint_dir, config_file_name, model_file_name, output_hf_checkpoint_dir):
+        self.initial_model = None
+        self.config_dict = None
+        self.components = None
+        self.checkpoint_dir = checkpoint_dir
+        self.config_file_name = config_file_name
+        self.model_file_name = model_file_name
+        self.output_hf_checkpoint_dir = output_hf_checkpoint_dir
+        self.registry = Registry(COMPONENTS)
+        self.component_factory = ComponentFactory(registry=self.registry)
+
+        self.checkpoint_dir = Path(self.checkpoint_dir)
+        config_file_path = checkpoint_dir / self.config_file_name
+        if not config_file_path.exists():
+            raise ValueError(f"Could not find {self.config_file_name} in {checkpoint_dir}")
+
+        # TODO resolve config hierarchically
+        self.config_dict = load_app_config_dict(config_file_path)
+        logging.info(f"Config\n{self.config_dict}")
+
+        self.checkpoint_path = checkpoint_dir / self.model_file_name
+
+    def convert_pytorch_to_hf_checkpoint(self):
+        # FIXME: make the conversion entry point configurable from outside:
+        # Which HuggingFaceAdapterConfig should be used etxactly currently it is too hard coded:
+        # Allow for a custom callable for conversion to HF to be given to the entrypoint
```

Comment:
This should be addressed. This code should not depend at all on the GPT2 specific implementation. Or, if we can only write it in a model specific form. This whole file should be moved to the GPT2 directory.
Ideally, a more generic and if necessary, abstract implementation would be implemented.

Reply:
Will be addressed with issue #71